tsweb/compress.go

```diff
@@ -0,0 +1,104 @@
+// Copyright (c) 2022 Tailscale Inc & AUTHORS All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package tsweb
+
+import (
+	"bufio"
+	"errors"
+	"io"
+	"net"
+	"net/http"
+
+	"github.com/andybalholm/brotli"
+)
+
+type compressingHandler struct {
```

Comment:
why not export this so people can opt into it fine-grained where they want it rather than only tsweb and its weirdo handler signature?